homeassistant/components/light/rfxtrx.py

```diff
@@ -0,0 +1,127 @@
+"""
+homeassistant.components.light.rfxtrx
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Rfxtrx lights.
+
+Configuration:
+
+To use Rfxtrx lights you will need to add the following to your
+configuration.yaml file.
+
+light:
+  platform: rfxtrx
+
+  devices:
+    ac09c4f1: Bedroom Light
+    ac09c4f2: Kitchen Light
+    ac09c4f3: Bathroom Light
+
+*Optional*
+
+  # Automatic add new light
+  automatic_add: True
+
+"""
+import logging
+import homeassistant.components.rfxtrx as rfxtrx
+import RFXtrx as rfxtrxmod
+
+from homeassistant.components.light import Light
+from homeassistant.util import slugify
+
+REQUIREMENTS = ['https://github.com/Danielhiversen/pyRFXtrx/archive/' +
+                'ec7a1aaddf8270db6e5da1c13d58c1547effd7cf.zip#RFXtrx==0.15']
+
+DOMAIN = "rfxtrx"
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Setup the RFXtrx platform. """
+    # Add light from config file
+    lights = []
+    devices = config.get('devices', None)
+    if devices:
+        for entity_id, entity_info in devices.items():
+            if entity_id not in rfxtrx.RFX_DEVICES:
+                _LOGGER.info("Add %s rfxtrx.light", entity_info['name'])
+                rfxobject = rfxtrx.get_rfx_object(entity_info['packetid'])
+                new_light = RfxtrxLight(entity_info['name'], rfxobject, False)
+                rfxtrx.RFX_DEVICES[entity_id] = new_light
+                lights.append(new_light)
+
+    add_devices_callback(lights)
+
+    def light_update(event):
+        """ Callback for sensor updates from the RFXtrx gateway. """
+        if isinstance(event.device, rfxtrxmod.LightingDevice):
```

Comment:
**Tip**: this is the perfect place to use a guard clause to prevent unnecessary indentation.
```python
if not isinstance(event.device, rfxtrxmod.LightingDevice):
    return

entity_id = …
```